avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -938,15 +939,24 @@ private void processEnum(final Schema schema, final JBlock body, FieldAction act
         JClass enumClass = schemaAssistant.classFromSchema(schema);
         newEnum = body.decl(enumClass, getUniqueName("enumValue"), JExpr._null());
 
-        JConditional ifBlock = null;
+        JSwitch switchBlock = body._switch(enumIndex);
         for (int i = 0; i < enumAdjustAction.adjustments.length; i++) {
-          JExpression ithVal =
-              schemaAssistant.getEnumValueByIndex(schema, JExpr.lit((Integer) enumAdjustAction.adjustments[i]),
-                  getSchemaExpr(schema));
-          JExpression condition = enumIndex.eq(JExpr.lit(i));
-          ifBlock = ifBlock != null ? ifBlock._elseif(condition) : body._if(condition);
-          ifBlock._then().assign((JVar) newEnum, ithVal);
+          JBlock caseBody = switchBlock._case(JExpr.lit(i)).body();
+          if (unknownEnumIndexes.contains(i)) {
+            caseBody._throw(JExpr._new(codeModel.ref(RuntimeException.class))
+                .arg(JExpr.lit(schema.getName() + " enum label impossible to deserialize: "
```

Comment:
can you pring the full schema name (with namespace) instead of the simple name?

Reply:
sure, changing to getFullName()